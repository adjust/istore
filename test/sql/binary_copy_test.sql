BEGIN;
-- binary_copy should copy data binary from country;
-- ./spec/binary_copy_spec.rb:22;
CREATE EXTENSION istore;
CREATE TABLE before (a istore);
INSERT INTO before values ('"1"=>"1"'),('"1"=>"2"'),('"1"=>"3"'),('"2"=>"1"'),('"2"=>NULL'),('"2"=>"2"');
CREATE TABLE after (a istore);
COPY before TO '/tmp/tst' WITH (FORMAT binary);
COPY after FROM '/tmp/tst' WITH (FORMAT binary);
SELECT * FROM after;
ROLLBACK;
