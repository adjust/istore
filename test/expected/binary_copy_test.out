-- bigistore binary_copy should copy data binary from country;
CREATE TABLE before_bigistore (a bigistore);
INSERT INTO before_bigistore values ('1=>1'),('1=>-9223372036854775807'),('1=>3'),('2=>1'),('2=>0'),('2=>9223372036854775806');
CREATE TABLE after_bigistore (a bigistore);
COPY before_bigistore TO '/tmp/tst' WITH (FORMAT binary);
COPY after_bigistore FROM '/tmp/tst' WITH (FORMAT binary);
SELECT * FROM after_bigistore;
              a              
-----------------------------
 "1"=>"1"
 "1"=>"-9223372036854775807"
 "1"=>"3"
 "2"=>"1"
 "2"=>"0"
 "2"=>"9223372036854775806"
(6 rows)

-- istore binary_copy should copy data binary from country;
CREATE TABLE before_istore (a istore);
INSERT INTO before_istore values ('1=>1'),('1=>-2147483647'),('1=>3'),('2=>1'),('2=>0'),('2=>2147483647');
CREATE TABLE after_istore (a istore);
COPY before_istore TO '/tmp/tst' WITH (FORMAT binary);
COPY after_istore FROM '/tmp/tst' WITH (FORMAT binary);
SELECT * FROM after_istore;
         a          
--------------------
 "1"=>"1"
 "1"=>"-2147483647"
 "1"=>"3"
 "2"=>"1"
 "2"=>"0"
 "2"=>"2147483647"
(6 rows)

